Docker build fails with CMake error

Running `docker build -t nektar-driftwave .` from root of repository on current tip of  `master` (2bb8304e15cfea20dd09960358ea160e19c750b6) fails with a CMake error in the build stage
```
RUN cd /root/tmp && mkdir build && cd build && \
    cmake -DNektar++_DIR=$INSTALL_PREFIX/lib64/nektar++/cmake -DCMAKE_INSTALL_PREFIX=$INSTALL_PREFIX/bin .. && \
    make install && cd /root && rm -Rf /root/tmp
```
with specific error message

```
[ 66%] Building CXX object CMakeFiles/DriftWaveSolver.dir/src/DriftWaveSystem.cpp.o
In file included from /root/tmp/src/DriftWaveSystem.cpp:36:
/root/tmp/src/DriftWaveSystem.h:114:18: error: 'virtual void Nektar::DriftWaveSystem::v_InitObject(bool)' marked 'override', but does not override
     virtual void v_InitObject(bool DeclareField = true) override;
                  ^~~~~~~~~~~~
/root/tmp/src/DriftWaveSystem.cpp: In member function 'virtual void Nektar::DriftWaveSystem::v_InitObject(bool)':
/root/tmp/src/DriftWaveSystem.cpp:58:47: error: no matching function for call to 'Nektar::DriftWaveSystem::v_InitObject(bool&)'
     AdvectionSystem::v_InitObject(DeclareField);
                                               ^
In file included from /root/tmp/src/DriftWaveSystem.h:39,
                 from /root/tmp/src/DriftWaveSystem.cpp:36:
/usr/local/include/nektar++/SolverUtils/AdvectionSystem.h:54:38: note: candidate: 'virtual void Nektar::SolverUtils::AdvectionSystem::v_InitObject()'
     SOLVER_UTILS_EXPORT virtual void v_InitObject();
                                      ^~~~~~~~~~~~
/usr/local/include/nektar++/SolverUtils/AdvectionSystem.h:54:38: note:   candidate expects 0 arguments, 1 provided
make[2]: *** [CMakeFiles/DriftWaveSolver.dir/build.make:76: CMakeFiles/DriftWaveSolver.dir/src/DriftWaveSystem.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:73: CMakeFiles/DriftWaveSolver.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
```

This appears to be related to #4 which highlights newer Nektar++ versions require an update to the `v_InitObject` function prototype signature. As I commented there this appears to have been updated on current `master` in 08fbfc0dff38efed16772df4ea6decb76b8319b0. The hash specified for the base `nektarpp/nektar-dev` Docker image in `Dockerfile` however appears to have been unchanged since 1b69039faa2e1c9794693520d17e286a258cc9b0 three years ago. Changing to `nektarpp/nektar-dev:latest` I get a different CMake error  
```
[ 33%] Building CXX object CMakeFiles/DriftWaveSolver.dir/src/DriftWaveSolver.cpp.o
In file included from /usr/local/include/nektar++/LibUtilities/BasicUtils/FieldIOXml.h:41,
                 from /usr/local/include/nektar++/LibUtilities/BasicUtils/PtsIO.h:49,
                 from /usr/local/include/nektar++/SolverUtils/EquationSystem.h:42,
                 from /usr/local/include/nektar++/SolverUtils/Driver.h:43,
                 from /root/tmp/src/DriftWaveSolver.cpp:36:
/usr/local/include/nektar++/LibUtilities/BasicUtils/Filesystem.hpp:43:21: error: 'filesystem' is not a namespace-name
   43 | namespace fs = std::filesystem;
      |                     ^~~~~~~~~~
In file included from /usr/local/include/nektar++/LibUtilities/BasicUtils/FieldIOXml.h:41,
                 from /usr/local/include/nektar++/LibUtilities/BasicUtils/PtsIO.h:49,
                 from /usr/local/include/nektar++/SolverUtils/EquationSystem.h:42,
                 from /usr/local/include/nektar++/SolverUtils/Driver.h:43,
                 from /root/tmp/src/DriftWaveSolver.cpp:36:
/usr/local/include/nektar++/LibUtilities/BasicUtils/Filesystem.hpp:56:46: error: 'fs' does not name a type; did you mean 'ffs'?
   56 | static inline std::string PortablePath(const fs::path &path)
      |                                              ^~
      |                                              ffs
/usr/local/include/nektar++/LibUtilities/BasicUtils/Filesystem.hpp:56:55: error: expected unqualified-id before '&' token
   56 | static inline std::string PortablePath(const fs::path &path)
      |                                                       ^
/usr/local/include/nektar++/LibUtilities/BasicUtils/Filesystem.hpp:56:54: error: expected ')' before '&' token
   56 | static inline std::string PortablePath(const fs::path &path)
      |                                       ~              ^~
      |                                                      )
/usr/local/include/nektar++/LibUtilities/BasicUtils/Filesystem.hpp:56:56: error: expected initializer before 'path'
   56 | static inline std::string PortablePath(const fs::path &path)
      |                                                        ^~~~
/usr/local/include/nektar++/LibUtilities/BasicUtils/Filesystem.hpp:69:15: error: 'fs' does not name a type; did you mean 'ffs'?
   69 | static inline fs::path UniquePath(std::string specFileStem)
      |               ^~
      |               ffs
make[2]: *** [CMakeFiles/DriftWaveSolver.dir/build.make:82: CMakeFiles/DriftWaveSolver.dir/src/DriftWaveSolver.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:95: CMakeFiles/DriftWaveSolver.dir/all] Error 2
make: *** [Makefile:149: all] Error 2
```
which suggests there are further changes required here for this to build against latest Nektar++.

Presumably there is some intermediate Nektar++ version this does successfully build against? All the GitHub Actions workflows runs which are still recorded are showing as failing due to an authentication error, so not much to go on there.

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,4 +1,4 @@
-FROM nektarpp/nektar-dev:2510247d
+FROM nektarpp/nektar-dev:v5.5.0
 
 ARG INSTALL_PREFIX=/usr/local
 

